cli/crates/federated-dev/src/dev/admin/types.rs

```diff
@@ -0,0 +1,38 @@
+use async_graphql::{InputObject, SimpleObject};
+use url::Url;
+
+#[derive(InputObject, Clone, Debug)]
+pub(crate) struct Header {
+    key: String,
+    value: String,
+}
+
+impl Header {
+    pub fn key(&self) -> &str {
+        &self.key
+    }
+
+    pub fn value(&self) -> &str {
+        &self.value
+    }
+}
+
+#[derive(InputObject)]
+pub(crate) struct PublishSubgraphInput {
+    pub(crate) name: String,
+    pub(crate) url: Url,
+    pub(crate) headers: Vec<Header>,
+}
+
+#[derive(SimpleObject)]
+pub(crate) struct PublishSubgraphSuccess {
+    __typename: &'static str,
```

Comment:
Quite odd that you're defining this - `__typename` implicitly exists on all GraphQL composite outputs.

Could you just do `struct PublishSubgraphSuccess` and save yourself some effort?

Reply:
async-graphql expects at least one field in the object. I don't really want to return anything if successful, so I wonder what we could have here?

Reply:
Could just return a `boolean` or similar rather than a whole object?

Reply:
Ah, it was late yesterday, of course...